stackslib/src/net/relay.rs

```diff
@@ -711,10 +713,43 @@ impl Relayer {
             &block.header.block_hash()
         );
 
+        let Some(canonical_block_header) =
+            NakamotoChainState::get_canonical_block_header(chainstate.db(), &sortdb)?
```

Comment:
I think this shouldn't be the `canonical_block_header`. It should be the last stacks block of the prepare phase or the parent of the stacks block being considered.

I don't think this needs to block this PR, though, so I opened an issue (#4109) -- can you just add a todo comment, with the issue number in it?

Reply:
Will do!